core/src/main/java/org/wildfly/channel/ChannelRecorder.java

```diff
@@ -17,13 +17,26 @@
 package org.wildfly.channel;
 
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 class ChannelRecorder {
     private ConcurrentHashMap<String, Stream> streams = new ConcurrentHashMap<>();
 
-    void recordStream(String groupId, String artifactId, String version) {
-        streams.putIfAbsent(groupId + ":" + artifactId + ":" + version, new Stream(groupId, artifactId, version, null));
+    void recordStream(String groupId, String artifactId, String version, String extension, String classifier, String sha256Checksum) {
```

Comment:
should we add an option for the recorder to calculate the hash if it's not present? So that we can generate a manifest with integrity protection